src/ngircd/irc-channel.c

```diff
@@ -236,8 +237,56 @@ join_forward(CLIENT *Client, CLIENT *target, CHANNEL *chan,
 	}
 
 	/* tell users in this channel about the new client */
+		#define IMPLEMENT_EXTENDED_JOIN
+	#ifndef IMPLEMENT_EXTENDED_JOIN
 	IRC_WriteStrChannelPrefix(Client, chan, target, false,
 				  "JOIN :%s",  channame);
+	#endif
+	#ifdef IMPLEMENT_EXTENDED_JOIN
+	CLIENT *c;
+	// Pointless alias
+	CHANNEL *Chan;
+	CONN_ID conn;
+	CL2CHAN *cl2chan;
+	Chan = chan;
+	cl2chan = Channel_FirstMember( Chan );
+	while(cl2chan) {
+		c = Channel_GetClient( cl2chan );
+		//if (!Remote) {
+			if (Client_Conn(c) <= NONE)
+				c = NULL;
+			else if(Client_Type(c) == CLIENT_SERVER)
+				c = NULL;
+		//}
+		if(c)
+			c = Client_NextHop(c);
+
+		if(c && c != Client) {
+			/* Ok, another Client */
+			conn = Client_Conn(c);
+			/*if (Client_Type(c) == CLIENT_SERVER)
```

Comment:
Please remove dead code and don't comment it out if it is no longer needed.
And don't forget to fix the indentation as well! ;-)